### PR TITLE
Fix http_build_query encoding array values with numeric keys

### LINKED ISSUE #153: Error 500 with filters
When querying with a filter such as: 
```php
$client = new Redmine\Client('http://redmine.example.com', 'API_ACCESS_KEY');
$filter = array(
	"f"  => array("spent_on"),
	"op" => array("spent_on" => "><"),
	"v"  => array("spent_on" => array(
			"2016-01-18",
			"2016-01-22"
		)
	),
);
$timeEntries = $client->time_entry->all($filter);
```

I receive a 500 Error and the Redmine log has
```
Completed 500 Internal Server Error in 47ms (ActiveRecord: 1.3ms)

NoMethodError (undefined method `to_sym' for ["0", "project"]:Array):
  app/models/query.rb:464:in `block in column_names='
  app/models/query.rb:464:in `collect'
  app/models/query.rb:464:in `column_names='
  app/models/query.rb:248:in `build_from_params'
  app/models/time_entry_query.rb:130:in `build_from_params'
  app/models/query.rb:254:in `build_from_params'
  app/controllers/timelog_controller.rb:43:in `index'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

After debugging I discovered the `PHP Redmine API` sends the filter numeric index's 
e.g
`limit=25&offset=0&f[0]=spent_on&op[spent_on]=><&v[spent_on][0]=2016-01-18&v[spent_on][1]=2016-01-22`
While Redmine expects the query string to be constructed without them
`limit=25&offset=0&f[]=spent_on&op[spent_on]=><&v[spent_on][]=2016-01-18&v[spent_on][]=2016-01-22`

This is caused by php [http_build_query](http://php.net/manual/en/function.http-build-query.php).

The fix is to construct you query string manually.

### PR DESCRIPTION
**Pull request by cjsewell**
Fixes #153

### CODE REVIEW

File: lib/Redmine/Api/AbstractApi.php [hunk 1]
```diff
@@ -117,6 +117,36 @@ protected function sanitizeParams(array $defaults, array $params)
...
+
+    /**
+     * Generate a Redmine compatible URL-encoded query string
+     * 
+     * @param mixed $data
+     * @param string $parentKey
+     * @param bool $isNested
+     * @return string string a URL-encoded string.
+     */
+    function httpBuildQuery($data, $parentKey = '', $isNested = false)
```
Comment (kbsali) [inline lib/Redmine/Api/AbstractApi.php]: Thanks for this proposal, although i'm not very keen at re-inventing the wheel! 
How about trying to add sort of a "potHttpBuildQuery" processor instead?
see this comment: http://php.net/manual/en/function.http-build-query.php#111819

If you can confirm that this 1 liner fixed your issue I would go in favor of that!

Comment (cjsewell) [inline lib/Redmine/Api/AbstractApi.php]: Yep, thats how I was originally going to implement   the fix, however I believe it's a bit unsafe.

For example, you may want to search for an item which has [123] in its title etc?

Comment (kbsali) [inline lib/Redmine/Api/AbstractApi.php]: in that case the [ and ] would be encoded and i guess it should work just fine.

Comment (cjsewell) [inline lib/Redmine/Api/AbstractApi.php]: Ahh OK, fair enough! Would you like me to update the PR with the one regex version?

Comment (kbsali) [inline lib/Redmine/Api/AbstractApi.php]: that'd be awesome yes! :)

Comment (cjsewell) [inline lib/Redmine/Api/AbstractApi.php]: Updated @kbsali!

**Comment by kbsali** [comment]
cheers @cjsewell !

**Pull request merged**